lib/util/string.ts

```diff
@@ -14,11 +14,13 @@ export function replaceAt(
   oldString: string,
   newString: string
 ): string {
-  return (
-    content.substring(0, index) +
-    newString +
-    content.substring(index + oldString.length)
-  );
+  const firstPart = content.slice(0, index);
+  const secondPart = content.slice(index);
+  if (secondPart.startsWith(oldString)) {
+    return firstPart + secondPart.replace(oldString, newString);
+  }
+
+  return content;
```

Comment:
why does the original code doesn't successfully replace, the new code looks more worse.

Reply:
The old implementation was wrong because first it did not check if the string exists and it made assumptions about the length of the string it though it was replacing. 

This code makes sure that the string we are trying to replace actually exists at the index we specified. And replace() only replaces the first instance it finds, so if the string starts with it, it only replaces what was meant to be replaced.

Hope that makes sense.

Reply:
An alternative would be to use the 'old' code and add a check to make sure that the string we want to replace does exist. Would that be more acceptable? 

In any case, the test i added should prevent the reported issue in the future. And also make sure i did not break anything else.

Reply:
I reverted all the changes in this file. Turns out  i could fix the bug elsewhere.